retinanet/losses.py

```diff
@@ -45,18 +45,41 @@ def forward(self, classifications, regressions, anchors, annotations):
 
             bbox_annotation = annotations[j, :, :]
             bbox_annotation = bbox_annotation[bbox_annotation[:, 4] != -1]
+            
+            classification = torch.clamp(classification, 1e-4, 1.0 - 1e-4)
 
             if bbox_annotation.shape[0] == 0:
                 if torch.cuda.is_available():
-                    regression_losses.append(torch.tensor(0).float().cuda())
-                    classification_losses.append(torch.tensor(0).float().cuda())
-                else:
+                    targets = torch.zeros(classification.shape)
+                    alpha_factor = torch.ones(targets.shape).cuda() * alpha
+
+                    alpha_factor = 1. - alpha_factor
+                    focal_weight = classification
+                    focal_weight = alpha_factor * torch.pow(focal_weight, gamma)
+
+                    bce = -((1.0 - targets) * torch.log(1.0 - classification))
+
+                    # cls_loss = focal_weight * torch.pow(bce, gamma)
+                    cls_loss = focal_weight * bce
+                    classification_losses.append(cls_loss.sum(), min=1.0)
```

Comment:
still here, it should be `classification_losses.append(cls_loss.sum())`, right?

Reply:
Yep